Gitlab/two-stage-pipeline-template/{{cookiecutter.outputDir}}/assume-role.sh

```diff
@@ -1,21 +1,33 @@
 #!/bin/bash
 ROLE=$1
 SESSION_NAME=$2
+PERMISSIONS_PROVIDER=$3
 
 unset AWS_SESSION_TOKEN
 export AWS_ACCESS_KEY_ID=$PIPELINE_USER_ACCESS_KEY_ID
 export AWS_SECRET_ACCESS_KEY=$PIPELINE_USER_SECRET_ACCESS_KEY
 
-cred=$(aws sts assume-role --role-arn "$ROLE" \
-                           --role-session-name "$SESSION_NAME" \
-                           --query '[Credentials.AccessKeyId,Credentials.SecretAccessKey,Credentials.SessionToken]' \
-                           --output text)
+if [ "$PERMISSIONS_PROVIDER" = "OpenID Connect" ]
+then
+    # Use assume-role-with-web-identity instead of assume-role
+    cred=$(aws sts assume-role-with-web-identity --role-arn "$ROLE" \
+                            --role-session-name "$SESSION_NAME" \
+                            --web-identity-token $CI_JOB_JWT_V2 \
```

Comment:
Should be quoted to avoid word splitting. Please run `shellcheck` on all scripts:

```
brew install shellcheck
shellcheck *.sh
```